tests/envs/test_conveyorbelt_env.py

```diff
@@ -0,0 +1,115 @@
+"""Unit tests for the ConveyorBelt environment.
+
+This module tests the functionality of the ConveyorBeltEnv class, including:
+- Valid state transitions based on discrete actions (no-op, shift left, shift right)
+- Proper video recording during environment interaction
+- Correct insertion and shifting behavior of box values on the conveyor belt
+"""
+
+from pathlib import Path
+
+import numpy as np
+import pytest
+from gymnasium.wrappers import RecordVideo
+
+from hybrid_failure_discovery.envs.conveyorbelt_env import (
+    ConveyorBeltAction,
+    ConveyorBeltEnv,
+    ConveyorBeltState,
+)
+
+
+def test_conveyorbelt_env_with_state_checks():
+    """Detailed test of ConveyorBeltEnv with video recording and state
+    transition checks."""
+
+    env = ConveyorBeltEnv(seed=42)
+
+    video_dir = Path("videos/test-conveyorbelt-env")
+    video_dir.mkdir(parents=True, exist_ok=True)
+
+    wrapped_env = RecordVideo(
+        env,
+        str(video_dir),
+        episode_trigger=lambda episode_id: episode_id == 0,
+    )
+
+    # Reset environment and get initial state
+    state, _ = wrapped_env.reset(seed=42)
+    assert isinstance(state, ConveyorBeltState)
+    current_values = state.values.copy()
+
+    print(f"\nInitial state values: {current_values}")
+
+    rng = np.random.default_rng(42)
+
+    for step in range(10):
+        # Sample random action
+        action_index = int(rng.integers(low=0, high=len(wrapped_env.action_space)))
+        action = ConveyorBeltAction(index=action_index)
+
+        # Step the environment
+        next_state, _, done, truncated, _ = wrapped_env.step(action)
+
+        assert isinstance(next_state, ConveyorBeltState)
+
+        next_values = next_state.values
+
+        print(f"\nStep {step + 1}")
```

Comment:
let's remove these prints in the test to keep the outputs clean. for anything you want to check, add an assertion instead